app/views.py

```diff
@@ -244,6 +246,34 @@ def recent_users():
                               'results': users,
                               'query': ''})
 
+@views.route('/activity', methods=['GET', 'POST'])
+@login_required
+def activity():
+    '''
+    View for the activity feed of recent events.
+    '''
+
+    events = Event.query_in_deployment().order_by(desc(Event.created_at)).limit(10).all()
+    shared_message_form = SharedMessageForm()
+
+    if request.method == 'POST':
+        if not current_user.display_in_search:
+            flash(gettext(u'We need your name before you can post a message.'), 'error')
```

Comment:
More generally, I think this if/else statement and associated test aren't necessary --we should handle this error situation elsewhere, not at message creation time.